### PR TITLE
Update to edition 2021, lint fixes, and other general code improvements

### PR DESCRIPTION
**Pull request by ChefKissInc**
Improvements include, but are not limited to, fixed typos and grammar, and similar

### CODE REVIEW

**Comment by ChefKissInc** [comment]
I'm still getting from my usage:
```
error: binding to `_` prefixed variable with no side-effect
  --> src/tables/madt/ic/ioapic.rs:50:5
   |
50 |     pub vector: u8,
   |     ^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#no_effect_underscore_binding
   ```
   And I have no idea why

**Comment by ChefKissInc** [comment]
All good now on my part

**Comment by ChefKissInc** [comment]
Okay, now it's actually 100% ready to be merged

File: .rustfmt.toml [hunk 1]
```diff
@@ -57,8 +57,6 @@ skip_children = false
 hide_parse_errors = false
 error_on_line_overflow = false
 error_on_unformatted = false
-report_todo = "Always"
-report_fixme = "Always"
```
Comment (Robbepop) [inline .rustfmt.toml:61]: Why remove this?

File: impl/src/bitfield/config.rs [hunk 2]
```diff
@@ -13,17 +13,17 @@ use syn::parse::Result;
...
-    pub filled: Option<ConfigValue<bool>>,
-    pub repr: Option<ConfigValue<ReprKind>>,
-    pub derive_debug: Option<ConfigValue<()>>,
-    pub derive_specifier: Option<ConfigValue<()>>,
+    pub bytes: Option<Value<usize>>,
+    pub bits: Option<Value<usize>>,
+    pub filled: Option<Value<bool>>,
+    pub repr: Option<Value<ReprKind>>,
+    pub derive_debug: Option<Value<()>>,
+    pub derive_specifier: Option<Value<()>>,
```
Comment (Robbepop) [inline impl/src/bitfield/config.rs:21]: Those renamings are very subjective. I personally liked `ConfigValue` more since it is more on point.

**Review by Robbepop — COMMENTED** [review]
Thanks a lot for the spelling fixes, edition upgrade, formatting updates and general changes.
I bet this was a lot of work! Many people and projects are depending on this old crate and it could seriously need some love.

The problem I am having with this PR is that it is pretty much unreviewable in its current form since many different kinds of works have been intertwined.
While scimming over the PR I saw a few problems, too, such as replacing `::core::primitves::usize` with just `usize` which is bad for proc. macro hygiene.

I currently do not really maintain this crate and I really feel sorry for all the unreviewed PRs and improvements at the moment. I may come back to this project but at the moment I do not have enough free time to work on it.
If you feel like you want to improve this project and are still in need of an interesting side project to work on I'd propose you to fork this repo and implement those improvements on your fork. If you succeed and continue improving the crate I will certainly post a recommendation to use your crate instead of this one or even consider transitioning ownership rights to you directly. What do you think?

File: .rustfmt.toml (hunk 1 above)
Comment (ChefKissInc) [reply .rustfmt.toml:61]: These keys have been removed according to rustfmt

File: impl/src/bitfield/config.rs (hunk 2 above)
Comment (ChefKissInc) [reply impl/src/bitfield/config.rs:21]: To be honest, me too, but clippy was complaining

**Comment by ChefKissInc** [comment]
> Thanks a lot for the spelling fixes, edition upgrade, formatting updates and general changes. I bet this was a lot of work! Many people and projects are depending on this old crate and it could seriously need some love.
> 
> The problem I am having with the PR is that it is not easy to review in its current form since many different kinds of works have been intertwined. While scimming over the PR I saw a few problems, too, such as replacing `::core::primitves::usize` with just `usize` which is bad for proc. macro hygiene.
> 
> I currently do not really maintain this crate and I really feel sorry for all the unreviewed PRs and improvements at the moment. I may come back to this project but at the moment I do not have enough free time to work on it. If you feel like you want to improve this project and are still in need of an interesting side project to work on I'd propose you to fork this repo and implement those improvements on your fork. If you succeed and continue improving the crate I will certainly post a recommendation to use your crate instead of this one or even consider transitioning ownership rights to you directly. What do you think?

Personally, if I were to own such bitfield crate, I wouldn't write it the way it's currently written. I keep finding more and more lints to fix and generally had a lot of trouble navigating the source code of this crate

File: .rustfmt.toml (hunk 1 above)
Comment (ChefKissInc) [reply .rustfmt.toml:61]: See here: rust-lang/rustfmt#5101

**Comment by Robbepop** [comment]
> > Thanks a lot for the spelling fixes, edition upgrade, formatting updates and general changes. I bet this was a lot of work! Many people and projects are depending on this old crate and it could seriously need some love.
> > The problem I am having with the PR is that it is not easy to review in its current form since many different kinds of works have been intertwined. While scimming over the PR I saw a few problems, too, such as replacing `::core::primitves::usize` with just `usize` which is bad for proc. macro hygiene.
> > I currently do not really maintain this crate and I really feel sorry for all the unreviewed PRs and improvements at the moment. I may come back to this project but at the moment I do not have enough free time to work on it. If you feel like you want to improve this project and are still in need of an interesting side project to work on I'd propose you to fork this repo and implement those improvements on your fork. If you succeed and continue improving the crate I will certainly post a recommendation to use your crate instead of this one or even consider transitioning ownership rights to you directly. What do you think?
> 
> Personally, if I were to own such bitfield crate, I wouldn't write it the way it's currently written. I keep finding more and more lints to fix and generally had a lot of trouble navigating the source code of this crate

Yeah it is an old codebase already. If you are looking for a nice side project to work on this might be a good chance to own your own fork and develop the space of Rust bitfields further for the ecosystem. :)

**Comment by ChefKissInc** [comment]
> > > Thanks a lot for the spelling fixes, edition upgrade, formatting updates and general changes. I bet this was a lot of work! Many people and projects are depending on this old crate and it could seriously need some love.
> > > The problem I am having with the PR is that it is not easy to review in its current form since many different kinds of works have been intertwined. While scimming over the PR I saw a few problems, too, such as replacing `::core::primitves::usize` with just `usize` which is bad for proc. macro hygiene.
> > > I currently do not really maintain this crate and I really feel sorry for all the unreviewed PRs and improvements at the moment. I may come back to this project but at the moment I do not have enough free time to work on it. If you feel like you want to improve this project and are still in need of an interesting side project to work on I'd propose you to fork this repo and implement those improvements on your fork. If you succeed and continue improving the crate I will certainly post a recommendation to use your crate instead of this one or even consider transitioning ownership rights to you directly. What do you think?
> > 
> > 
> > Personally, if I were to own such bitfield crate, I wouldn't write it the way it's currently written. I keep finding more and more lints to fix and generally had a lot of trouble navigating the source code of this crate
> 
> Yeah it is an old codebase already. If you are looking for a nice side project to work on this might be a good chance to own your own fork or implementation (no need for fork if you rewrite everything anyways) and develop the space of Rust bitfields further for the ecosystem. :)

Yeah, I'm going to write a new crate, I just needed to fix this one temporarily, and thought I'd contribute these changes for anyone that uses this too

**Pull request closed without merging**